plugins/videobasedtracker/Types.h

```diff
@@ -0,0 +1,81 @@
+/** @file
+    @brief Header
+
+    @date 2015
+
+    @author
+    Sensics, Inc.
+    <http://sensics.com/osvr>
+*/
+
+// Copyright 2015 Sensics, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef INCLUDED_Types_h_GUID_819757A3_DE89_4BAD_3BF5_6FE152F1EA08
+#define INCLUDED_Types_h_GUID_819757A3_DE89_4BAD_3BF5_6FE152F1EA08
+
+// Internal Includes
+// - none
+
+// Library/third-party includes
+#include <opencv2/features2d/features2d.hpp>
+
+// Standard includes
+#include <vector>
+#include <list>
+#include <string>
+#include <memory>
+
+namespace osvr {
+namespace vbtracker {
+    class Led;
+    class LedIdentifier;
+    class BeaconBasedPoseEstimator;
+
+    /// @todo Replace usages of this with Eigen or cv matrices.
+    typedef std::vector<std::vector<double> > DoubleVecVec;
+
+    typedef std::vector<std::string> PatternStringList;
+
+    /// @todo std::list used here for ease of rotate, but has terrible data
+    /// locality - consider changing when a more efficient rotation-invariant
+    /// string match algorithm is used.
+    typedef std::list<bool> LedPattern;
+    typedef std::vector<LedPattern> PatternList;
+
+    typedef std::vector<cv::KeyPoint> KeyPointList;
+    typedef KeyPointList::iterator KeyPointIterator;
+
+    typedef float Brightness;
+    typedef std::list<Brightness> BrightnessList;
+    typedef std::pair<Brightness, Brightness> BrightnessMinMax;
+
+    typedef std::unique_ptr<BeaconBasedPoseEstimator> EstimatorPtr;
+    typedef std::unique_ptr<LedIdentifier> LedIdentifierPtr;
+
+    typedef std::list<Led> LedGroup;
+
+    /// @name Containers of "per-sensor" objects
+    /// @brief It seems like in a "well-formed" video-based tracker, there is
```

Comment:
Its current state is a combination of what was needed to be able to hand to OpenCV, pushing to get something working, and having abandoned the Oculus code in the middle of the project when I realized that we weren't going to be able to configure its camera for synchronous operation.  The Oculus code works enough to get the camera data and keep the DK2 alive (which turns out to be required to do my DirectMode development, so we may want to pull that code out separately).

My guess about being commented out is that only one of the three was needed to make the code run.